std/file.d

```diff
@@ -4458,11 +4458,16 @@ void rmdirRecurse(ref DirEntry de)
     }
     else
     {
-        // all children, recursively depth-first
-        foreach (DirEntry e; dirEntries(de.name, SpanMode.depth, false))
-        {
-            attrIsDir(e.linkAttributes) ? rmdir(e.name) : remove(e.name);
-        }
+        // dirEntries is @system because it uses a DirIterator with a
+        // RefCounted variable, but here, no references to the payload is
+        // escaped to the outside, so this should be @trusted
+        () @trusted {
+            // all children, recursively depth-first
+            foreach (DirEntry e; dirEntries(de.name, SpanMode.depth, false))
+            {
+                attrIsDir(e.linkAttributes) ? rmdir(e.name) : remove(e.name);
+            }
+        }();
```

Comment:
can this be done as 
```d
// all children, recursively depth-first
foreach (DirEntry e; () @trusted { return dirEntries(de.name, SpanMode.depth, false); } ())
{
    attrIsDir(e.linkAttributes) ? rmdir(e.name) : remove(e.name);
}
```
?

Reply:
No. That's not @safe . The `DirIterator` dtor which is @system is called outside the @trusted zone, when the reference is destroyed. And it will discard the whole purpose of this because the payload reference is escaped from the wrapped @trusted zone anyway, so it's preferable to wrap the whole foreach.

Reply:
OK.